### PR TITLE
prov/verbs: Use common fork_unsafe variable instead

### PR DESCRIPTION
**Pull request by gladkovdmitry17**
Signed-off-by: Dmitry Gladkov <dmitry.gladkov@intel.com>

### CODE REVIEW

**Comment by gladkovdmitry17** [comment]
@shefty @a-ilango What do you think about? Does this make sense?

**Comment by a-ilango** [comment]
Yes. Btw, looking at the help strings they seem to make opposite assumptions. The one in common code assumes fork_unsafe is no by default but the one here assumes its yes by default.

**Comment by gladkovdmitry17** [comment]
Looking [at the place](https://github.com/ofiwg/libfabric/pull/3528/files#diff-0b76fd023d10e5c3d6bacced0c3de5a4R52) where it is initiated by default value, this parameter is 0 (i.e. "no").  Seems the string definition was unclear.

**Comment by gladkovdmitry17** [comment]
@a-ilango I've added commit to the PR that removes mention of this variable from the fi_verbs(7) man page

File: prov/verbs/src/fi_verbs.c [hunk 1]
```diff
@@ -618,16 +618,9 @@ static int fi_ibv_read_params(void)
...
-				  "support isn't needed - No need to use this option, "
-				  "because extra memory will be consumed when enabling "
-				  "fork suppport.",
-				  &fi_ibv_gl_data.fork_unsafe)) {
-		VERBS_WARN(FI_LOG_CORE,
-			   "Invalid value of fork_unsafe\n");
-		return -FI_EINVAL;
-	}
+
+	fi_param_get_bool(NULL, "fork_unsafe", &fi_ibv_gl_data.fork_unsafe);
```
Comment (a-ilango) [inline prov/verbs/src/fi_verbs.c]: The function returns int and we need to check for errors.

Comment (gladkovdmitry17) [reply prov/verbs/src/fi_verbs.c]: Sure,
I've updated the PR.

**Pull request merged**